### ISSUE TITLE
Direct Messages from one teams appear in all my teams

### ISSUE THREAD

**Issue by guatedude2** [mattermost/mattermost-server#7474]
#### Summary
When having a user in the direct message list from a specific team, switching to other teams that the user doesn't belong the user keeps appearing in the other teams even though they aren't part of the team.

#### Steps to reproduce
- Set the setting "Enable users to open Direct Message channels with:" to "Any member of the team" 
- Create two teams (Team A, Team B)
- Open a direct message with a user that is only on Team A.
-Switch to Team B, the user continues to appear in Team B even though they aren't in that team.

#### Expected behavior
When having a direct message open from the user in Team A and switching to Team B, the user should disappear from the list and show the last active conversation for that Team or show the Town Square channel.

#### Observed behavior (that appears unintentional) 
The user in Team A appears in the direct messages list from Team B

#### Possible fixes
I haven't dug in the code yet, I can take a look and investigate

**Comment by lindy65** [comment]
Hi @guatedude2 

Thank you for your feedback,

This is by design.  Direct messages are common per server, not per team so all users on the server will appear in direct messages no matter what team you are in on the server.

**Comment by guatedude2** [comment]
Its an odd behavior since the direct messages list that appears in the + sign doesn't show the same users.

**Comment by lindy65** [comment]
Hi @guatedude2 

Thanks for your feedback,

The `+` to the right of `DIRECT MESSAGES` should show the same users as clicking on `More...` at the foot of `DIRECT MESSAGES`.

These are users you can still open a direct message or group message with (so users who are already in your direct message list should not appear).

Does this help or am I misunderstanding your issue?

**Comment by guatedude2** [comment]
That's not happening on my end. Note that I have "Enable users to open Direct Message channels with" set to true
Here's a screen shot of the two teams: 
![screen shot 2017-09-19 at 2 14 43 pm](https://user-images.githubusercontent.com/918429/30608453-6b234d2a-9d46-11e7-91c2-c11462683531.png)
![screen shot 2017-09-19 at 2 14 52 pm](https://user-images.githubusercontent.com/918429/30608454-6b275f32-9d46-11e7-8b3a-33fdb6ebd550.png)

**Comment by lindy65** [comment]
Thanks for your feedback @guatedude2 

Could you share which version of Mattermost you are using?

Also, I was not correct when I said users who are already in your direct message list below `DIRECT MESSAGES` should not appear in the `Direct Messages` modal when you click on `+` or `More...`.  Users will still appear in this modal because they can still be added to group messages.  Sorry about that...

**Comment by lfbrock** [comment]
Hi @guatedude2,

To clarify - the setting in the System Console only hides the users from other teams in the Direct Messages modal UI (it's not blocked on the back end). 

This setting was added at Mattermost v3.0 so people who previously used the site (where accounts were per team) could continue to use it in a similar way even though accounts are now per server. 

We will likely replace the setting with something else in future. 

So to sum up - as Lindy mentioned, it's expected behaviour to have Direct Messages show up across teams. The experience with restricting the Direct Message list to only users on the team is a little strange, but that setting will likely be deprecated and replaced with something else in future.

**Comment by guatedude2** [comment]
That makes sense.

**Comment by lindy65** [comment]
Thanks @lfbrock @guatedude2 

I'll close this issue off for now...
